server/src/infra/ldap_handler.rs

```diff
@@ -2060,6 +2105,79 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_compare() {
```

Comment:
Add tests to cover `CompareFalse`, an invalid dn, an invalid attribute, and no such object?